src/wayland/drm_syncobj/mod.rs

```diff
@@ -0,0 +1,338 @@
+//! DRM syncobj protocol
+
+use std::{cell::RefCell, os::unix::io::AsFd};
+use wayland_protocols::wp::linux_drm_syncobj::v1::server::{
+    wp_linux_drm_syncobj_manager_v1::{self, WpLinuxDrmSyncobjManagerV1},
+    wp_linux_drm_syncobj_surface_v1::{self, WpLinuxDrmSyncobjSurfaceV1},
+    wp_linux_drm_syncobj_timeline_v1::{self, WpLinuxDrmSyncobjTimelineV1},
+};
+use wayland_server::{
+    protocol::wl_surface::WlSurface, Client, DataInit, Dispatch, DisplayHandle, GlobalDispatch, New,
+    Resource, Weak,
+};
+
+use super::compositor::{self, with_states, BufferAssignment, Cacheable, SurfaceAttributes};
+use crate::backend::drm::DrmDeviceFd;
+
+mod sync_point;
+pub use sync_point::{DrmSyncPoint, DrmTimeline};
+
+/// Handler trait for DRM syncobj protocol.
+pub trait DrmSyncobjHandler {
+    // TODO better way to deal with this?
+    /// DRM device for importing syncobj file descriptors
+    fn import_device(&self) -> &DrmDeviceFd;
+}
+
+/// Data associated with a drm syncobj global
+#[allow(missing_debug_implementations)]
+pub struct DrmSyncobjGlobalData {
+    filter: Box<dyn for<'c> Fn(&'c Client) -> bool + Send + Sync>,
+}
+
+/// Pending DRM syncobj sync point state
+#[derive(Debug, Default)]
+pub struct DrmSyncobjCachedState {
+    /// Timeline point signaled when buffer is ready to read
+    pub acquire_point: Option<DrmSyncPoint>,
+    /// Timeline point to be signaled when server is done with buffer
+    pub release_point: Option<DrmSyncPoint>,
+}
+
+impl Cacheable for DrmSyncobjCachedState {
+    fn commit(&mut self, _dh: &DisplayHandle) -> Self {
+        Self {
+            acquire_point: None,
+            release_point: None,
+        }
+    }
+
+    fn merge_into(self, into: &mut Self, _dh: &DisplayHandle) {
+        if self.acquire_point.is_some() && self.release_point.is_some() {
+            into.acquire_point = self.acquire_point;
+            into.release_point = self.release_point;
```

Comment:
In case we unblock a bunch of commits at once, we still need to make sure to signal the release_point of unused buffers.